docs/how-to/unified_memory.rst

```diff
@@ -0,0 +1,595 @@
+.. meta::
+  :description: This chapter describes introduces Unified Memory (UM) and shows
+                how to use it in AMD HIP.
+  :keywords: AMD, ROCm, HIP, CUDA, unified memory, unified, memory, UM, APU
+
+*******************************************************************************
+Unified Memory
+*******************************************************************************
+
+Concept
+=======
+
+In a conventional architectures, both CPUs a GPUs have a dedicated memory,
+the RAM and VRAM, respectively. Inherent in this architectural design is the
+need for continuous memory copying to allow the processors to access the
+appropriate data. This reduces the maximum available memory capacity and
+bandwidth. One way to avoid these effects is to use Heterogeneous System
+Architectures (HSA) such as Unified Memory.
+
+The Unified Memory (UM) can be understood as a single memory address space that
+is accessible from any processor within a system. This setup enables
+applications to allocate data that can be read or written by code running on
+either CPUs or GPUs. The Unified memory model is shown in the figure below.
+
+.. figure:: ../data/unified_memory/um.svg
+
+AMD Accelerated Processing Unit (APU) is a typical example of a Unified Memory
+Architecture. On a single die, a central processing unit (CPU) is combined with
+an integrated graphics processing unit (iGPU) and both have the access for a
+high bandwidth memory (HBM) module, named as Unified Memory. The CPU enables
+high-performance low latency operations, while the GPU is optimized for
+high-throughput (data processed by unit time).
+
+How-to use?
+===========
+
+Unified Memory Management (UMM) is a feature that can simplify the complexities
+of memory management in GPU computing. It is particularly useful in
+heterogeneous computing environments with heavy memory usage with both a CPU
+and a GPU which would require large memory transfers. Here are some areas where
+UMM can be beneficial:
+
+- **Simplification of Memory Management**:
```

Comment:
Recommended

Simplification of memory management

UMM can help to simplify the complexities of memory management. This can make it easier for developers to write code without worrying about memory allocation and deallocation details.

Reply:
Updated.